src/frame_decoder.rs

```diff
@@ -622,3 +629,73 @@ impl Read for FrameDecoder {
         }
     }
 }
+
+/// Decode multiple frames into the output slice.
+///
+/// `input` must contain an exact number of frames.
+///
+/// `output` must be large enough to hold the decompressed data. If you don't know
+/// how large the output will be, use [`FrameDecoder`] instead.
+///
+/// Returns the number of bytes written to `output`.
+pub fn decode_all(mut input: &[u8], mut output: &mut [u8]) -> Result<usize, FrameDecoderError> {
+    let mut total_bytes_written = 0;
+    while !input.is_empty() {
+        let mut decoder = FrameDecoder::new();
```

Comment:
The decoder can be reused so any scratch allocations are reused for the next frame

Reply:
Done.

Reply:
I added a commit that takes this a step further by changing these to be methods on `FrameDecoder`.